packages/js/data/src/crud/types.ts

```diff
@@ -13,3 +22,79 @@ export type Item = {
 export type ItemQuery = BaseQueryParams & {
 	[ key: string ]: unknown;
 };
+
+export type CrudActions< ResourceName, ItemConfig, MutableProperties > =
+	MapActions<
+		{
+			create: ( query: ItemQuery ) => Item;
+			update: ( query: ItemQuery ) => Item;
+		},
+		ResourceName,
+		MutableProperties,
+		ItemConfig
+	> &
+		MapActions<
+			{
+				delete: ( id: IdType ) => Item;
+			},
+			ResourceName,
+			IdType,
+			ItemConfig
+		>;
+
+export type CrudSelectors<
+	ResourceName,
+	PluralResourceName,
+	ItemConfig,
+	ItemQueryConfig,
```

Comment:
Is there a specific reason you used `Config` here? It kinda confused me initially. Would `Type` make more sense -> `ItemType` & `ItemQueryType`?

Reply:
I agree.  I was trying to avoid the ambiguity between the `Item` type that already exists and the one the data store passes in ("configures"), but not sure it landed well.

I still see some room for confusion using `ItemType` but it's probably better than `ItemConfig`.